### PR TITLE
Initial commit on #275 - Fix the ELB

### PR DESCRIPTION
**Pull request by tayzlor**
- Fix health check for ELB
- Expose health check port on haproxy container
- Use PROXY mode on ELB in AWS

This fix needs more testing

### CODE REVIEW

File: roles/haproxy/files/haproxy.tmpl [hunk 1]
```diff
@@ -1,15 +1,25 @@
...
+    # create a status URI in /haproxy_status that will return
+    # a 200 is backend is healthy, and 503 if it isn't. This
+    # URI is queried by the ELB.
+    acl backend_dead nbsrv(consul_backend) lt 1
+    monitor-uri /haproxy_status
+    monitor fail if backend_dead
 
 frontend www
-    bind *:80
+    bind *:80 accept-proxy
```
Comment (tayzlor) [inline roles/haproxy/files/haproxy.tmpl]: There appears to be a problem with this line here (when running up in the vagrant machine/aws, so removing it for now)

**Comment by wallies** [comment]
Does this need terraform 0.5.3, if so you should add that to the terraform version check

**Comment by tayzlor** [comment]
I ran it up on 0.5.2

**Comment by wallies** [comment]
Does it require 0.5.2 then, if so, can we add that check to the terraform version

**Comment by tayzlor** [comment]
I can't seem to test this fix properly. Here's what I tried, pasting here incase anyone else wants to have a go - 

- provisioned the instances
- got the ELB hostname and did - ```dig +short apollo-elb-578724806.eu-west-1.elb.amazonaws.com``` as exmaple. You should get the IP(s) of the ELB
- Tried running a ```curl -s -v -H 'Host: marathon.example.com' http://54.76.83.152```

My end goal was to get an IP and stick it in ```/etc/hosts``` and access that statically mapped host through my browser.

**Comment by pmbauer** [comment]
I discovered what the problem is.
https://github.com/Capgemini/Apollo/commit/33e53fb56a1c1486dad761428ba481ce4c05332e#diff-ee33aa395b84461181b557096d97ed52R8

The listener here needs to be 'http' protocol.
Then I was able to successfully run `curl -v -H 'Host: marathon.example.com' http://apollo-mesos-elb-XXXXXXX.us-west-1.elb.amazonaws.com`

I kept the health check listener the same as in 33e53fb.

**Pull request merged**